### ISSUE TITLE
GrpcService with default serialization formats does not show a debug form in DocService

### ISSUE THREAD

**Issue by trustin** [line/armeria#1134]
Unlike `THttpService`, a user has to enable more serialization formats to make `DocService` show the debug form for a `GrpcService`. Perhaps we should:

- make it less surprising,
- update our documentation, or
- add some instructions to the DocService page without the debug form

What do you think, @anuraaga ?

**Comment by anuraaga** [comment]
Yeah with some more documentation and with text in `DocService` or at least a `logger.info`, it would be much clearer. I think the default for it to be off is safer, at least for servers that might be exposed to external traffic since exposing unframed JSON can be a huge DDoS opportunity - it's the main reason I didn't default those on.

I have been thinking for some time, and a little related to this, perhaps we could have the concept of modes in armeria. Something along the lines of
- Internal vs external
- RPC vs REST
- Unary vs streaming
- ... etc (though not too many or it's just spaghetti soup)

which group together best-practice defaults / flags for each use case. In this case, I would default on in `Internal` and off in `external`. Defaults for things like the path parsing cache, timeouts, logging, and probably many things could also be tuned for these use cases.
